lib/newProfile.mli

```diff
@@ -50,3 +50,14 @@ val pause : unit -> accu option
 val resume : accu -> unit
 (** Profiling must not be active.
     Activates profiling with the given state. *)
+
+type sums
+(** Timings for sub-events *)
+
+val with_profiling : (unit -> 'a) -> MiniJson.t list * sums * 'a
+(** Runs the given function with profiling active and returns the
+    produced events and sum times of subevents. *)
```

Comment:
changed to also be compatible with an already-running profile (will be useful in later PR)